packages/sanity/src/core/releases/plugin/ReleasesStudioLayout.tsx

```diff
@@ -1,18 +1,8 @@
 import {type LayoutProps} from '../../config'
-import {BundlesProvider} from '../../store/bundles/BundlesProvider'
-import {AddonDatasetProvider} from '../../studio'
 
 export function ReleasesStudioLayout(props: LayoutProps) {
   // TODO: Replace for useReleasesEnabled
   const {enabled, mode} = {enabled: true, mode: 'default'}
 
-  if (!enabled) {
-    return props.renderDefault(props)
-  }
-
-  return (
-    <AddonDatasetProvider>
-      <BundlesProvider>{props.renderDefault(props)}</BundlesProvider>
-    </AddonDatasetProvider>
```

Comment:
We need to think on how to inject the `AddonDatasetProvider` into the studio when needed. If we remove it from here and the user decides to only include the `releases` plugin, but not `comments` or `tasks` the `useAddonDataset` hook will crash in https://github.com/sanity-io/sanity/pull/7077/files#diff-e6d5f6b7c44a215f6830735c676cbb63da8df1457870f3b44897293679dff703R283

Reply:
That's a great point, sorry for missing that.

I'm not very familiar with this part of the codebase yet, but do you think it would be plausible to move the addon dataset provider to the resource cache, too? That way we could always include it, but it would be lazily initialised once any plugin tried to access it.

Reply:
Now that I know how the resources cache works I like that idea!